serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -462,11 +472,13 @@ def generate(
         try:
             next_tokens = sample(logits, sampling_params, self.vocab_size)
             assert next_tokens is not None
-
             outputs = []
-            for i, (sequence_id, new_token) in enumerate(
-                zip(sequence_ids, next_tokens)
+            for i, (sequence_id, new_token, request) in enumerate(
+                zip(sequence_ids, next_tokens, requests)
             ):
+                if not new_token in request.sampling_params.appeared_tokens_freq:
+                    request.sampling_params.appeared_tokens_freq[new_token] = 0
+                request.sampling_params.appeared_tokens_freq[new_token] += 1
```

Comment:
We also need this logic in https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/model/paged_cache_model.py#L489-L499

This is the code path for the case when batched sampling fails for numerical reasons (negative prob). You can exercise this code path by running `benchmark_throughput.py` with 7B + small `min_decode_steps` and `max_decode_steps` (12 and 16, for example).

Reply:
I have just updated the code, have you tried the latest one?

Reply:
I will add it

Reply:
added